base/float.jl

```diff
@@ -463,16 +463,29 @@ for Ti in (Int8, Int16, Int32, Int64, Int128, UInt8, UInt16, UInt32, UInt64, UIn
                 $(Tf(typemin(Ti))-one(Tf)) < x < $(Tf(typemax(Ti))+one(Tf)) || throw(InexactError())
                 unsafe_trunc($Ti,x)
             end
-        else
+            @eval function convert(::Type{$Ti}, x::$Tf)
+                if ($(Tf(typemin(Ti))) <= x <= $(Tf(typemax(Ti)))) && (trunc(x) == x)
+                    return unsafe_trunc($Ti,x)
+                else
+                    throw(InexactError())
+                end
+            end
+        else #
```

Comment:
I can't remember, this was quite a long time ago (if only I had left a comment on what that comment was going to be...)